sdk/storage/azure-storage-blob/tests/test_helpers_async.py

```diff
@@ -60,3 +65,103 @@ async def read(self, size: int = -1) -> bytes:
         self._offset += len(data)
 
         return data
+
+
+class MockAioHttpClientResponse(ClientResponse):
+    def __init__(
+        self, url: str,
+        body_bytes: bytes,
+        headers: Optional[Dict[str, Any]] = None,
+        status: int = 200,
+        reason: str = "OK"
+    ) -> None:
+        super(MockAioHttpClientResponse).__init__()
+        self._url = url
+        self._body = body_bytes
+        self._headers = headers
+        self._cache = {}
+        self._loop = None
+        self.status = status
+        self.reason = reason
+
+
+class MockStorageTransport(AsyncHttpTransport):
+    async def send(self, request: HttpRequest, **kwargs: Any) -> RestAioHttpTransportResponse:
+        if request.method == 'GET':
```

Comment:
You need to handle content validation for GET. When `validate_content=True`, an additional header will be placed in the request, if (and only if) you see that header here, you need to include content_MD5 in the response.